pmlc/dialect/stripe/util.cc

```diff
@@ -35,6 +37,166 @@ void createMainParallelFor(mlir::FuncOp funcOp) {
   builder.create<TerminateOp>(funcOp.getLoc());
 }
 
+bool hasAttr(Operation* op, const std::string& attr) {
```

Comment:
Many of these utility functions are working around a design flaw I think.  Specifically, if the 'stripe attributes' were placed directly on the Op, rather than into a DictionaryAttr on the op, then things would be much easier.  For example, 'if (hasAttr(op, "name"))' would just be 'if(op->getAttr("name")' using existing MLIR support with no additional functions required.  Additionally, adding a new attribute would also be trivial, and not involve copying a while dictionary to make a change.  Let me think about what the right thing to do here is and also talk to Frank.

Reply:
The reason for the dictionary was to isolate stripe 'tags/attrs' from MLIR ones. If we're not going to bother with translating to stripe classic, then I agree, we can just put them directly on the op.

Reply:
Even if we do translate to/from stripe, the dictionary is pretty awkward, so perhaps something like either: 
- A name prefix (i.e. stripe_<some name>) or
- Assume any attribute not known to MLIR specifically should be translated back

But yes, this may all be a mute point if we end up going straight to extended affine...